api/v1alpha4/openstackcluster_webhook_test.go

```diff
@@ -71,6 +71,58 @@ func TestOpenStackCluster_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "OpenStackCluster.Spec.Bastion with mutable spec",
+			oldTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					CloudName: "foobar",
+					Bastion: &Bastion{
+						Instance: OpenStackMachineSpec{
+							CloudName: "foobar",
+							Image:     "foobar",
+							Flavor:    "minimal",
+						},
+						Enabled: false,
+					},
+				},
+			},
+			newTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					CloudName: "foobar",
+					Bastion: &Bastion{
+						Instance: OpenStackMachineSpec{
+							CloudName: "foobarbaz",
+							Image:     "foobarbaz",
+							Flavor:    "medium",
+						},
+						Enabled: true,
+					},
+				},
+			},
```

Comment:
This is an interesting one. I note that in this unit test we're changing both the bastion spec and the enabled flag. This is fine: we will create a new Bastion with the new spec.

However, note we also permit changing the bastion spec without flipping enabled, but this won't work. It won't fail, but the bastion will not be updated until the enabled flag is flipped to false and the bastion is deleted.

What do we want to do about this? Options:

1. Ignore it
2. Document it
3. Don't allow changing bastion spec if status.Bastion != nil
4. Make it work

For now I feel like 2 + 3 would be user friendly and relatively simple.

Reply:
I totally agree that we should document this behavior.
Regarding 3: do you think it's worth the effort to update our `reconcileBastion` to delete the bastion if sth has changed? Or is it enough to only allow changes if the enabled flag is false? Both are fine for me, I just think from the customer's perspective.
WDYT?

Reply:
I feel like automatically deleting and recreating the bastion if something changed would be the most user friendly, but I should strongly caveat that with: I don't feel like I understand all the potential use cases here. Have we firmly set an expectation that we can blow the bastion away? Nobody's running anything on the bastion which is going to suffer data loss? I'd hope not.

I also think it would be fine to do 3 and do 4 later. I don't think they're incompatible: if the user writes orchestration to delete and recreate the bastion if they need to change it, that will continue to work even if we remove the requirement to do it. We're also being nice to them in the meantime by pointing out that it doesn't work.

Reply:
Okay, I'll change my PR to 2 + 3 👍🏻
Thanks!

Reply:
I updated my PR, can you PTAL @mdbooth? 🙂 
Documentation will be pushed during the day.

Reply:
Added the documentation as well. Thx